pkg/rendering/renderer_proxy.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package rendering
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/kustomize/v3/pkg/resource"
+
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
+	mcoconfig "github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/util"
+)
+
+func (r *Renderer) newProxyRenderer() {
+	r.renderProxyFns = map[string]renderFn{
+		"Deployment":            r.renderProxyDeployment,
+		"Service":               r.renderNamespace,
+		"ServiceAccount":        r.renderNamespace,
+		"ConfigMap":             r.renderNamespace,
+		"ClusterRole":           r.renderClusterRole,
+		"ClusterRoleBinding":    r.renderClusterRoleBinding,
+		"Secret":                r.renderProxySecret,
+		"Role":                  r.renderNamespace,
+		"RoleBinding":           r.renderNamespace,
+		"Ingress":               r.renderNamespace,
+		"PersistentVolumeClaim": r.renderNamespace,
+	}
+}
+
+func (r *Renderer) renderProxyDeployment(res *resource.Resource) (*unstructured.Unstructured, error) {
+	u, err := r.renderDeployments(res)
+	if err != nil {
+		return nil, err
+	}
+	obj := util.GetK8sObj(u.GetKind())
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	crLabelKey := mcoconfig.GetCrLabelKey()
+	dep := obj.(*v1.Deployment)
+	dep.ObjectMeta.Labels[crLabelKey] = r.cr.Name
+	dep.Spec.Selector.MatchLabels[crLabelKey] = r.cr.Name
+	dep.Spec.Template.ObjectMeta.Labels[crLabelKey] = r.cr.Name
+	dep.Spec.Replicas = config.GetReplicas(config.RBACQueryProxy, r.cr.Spec.AdvancedConfig)
+
```

Comment:
Here we need to set `dep.Spec.Name`. You can get the correct name when this PR https://github.com/open-cluster-management/multicluster-observability-operator/pull/605 is merged.

Reply:
it was covered in #605 ignore this comment.